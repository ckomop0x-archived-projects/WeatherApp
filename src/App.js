import React, { Component } from 'react';
import "bootswatch/journal/bootstrap.css";
import { Navbar, NavItem, Nav, Grid, Row, Col } from "react-bootstrap";
import WeatherDisplay from './WeatherDisplay';

const PLACES = [
	{ name: "Amsterdam"},
	{ name: "Moscow"},
	{ name: "Kiev"},
	{ name: "Uzhhorod"},
	{ name: "Prague"},
	{ name: "Santa Cruz"},
	{ name: "Honolulu"}
];

class App extends Component {
	constructor() {
		super();
		this.state = {
			activePlace: 0,
		};
	}

	render() {
	  const activePlace = this.state.activePlace;

	  return (
      <div className="App">
	      <div>
		      <Navbar>
			      <Navbar.Header>
				      <Navbar.Brand>
					     Weather App
				      </Navbar.Brand>
			      </Navbar.Header>
		      </Navbar>
		      <Grid>
			      <Row>
				      <Col md={4} sm={4}>
					      <h3>Select a city</h3>
					      <Nav
						      bsStyle="pills"
						      stacked
						      activeKey={activePlace}
						      onSelect={index => {
							      this.setState({ activePlace: index });
						      }}
					      >
						      {PLACES.map((place, index) => (
							      <NavItem
								      key={index}
								      eventKey={index}
							      >{place.name}</NavItem>
						      ))}
					      </Nav>
				      </Col>
				      <Col md={8} sm={8}>
					      <WeatherDisplay
						      key={activePlace}
						      name={PLACES[activePlace].name} />
				      </Col>
			      </Row>
		      </Grid>
	      </div>
      </div>
    );
  }
}

export default App;
