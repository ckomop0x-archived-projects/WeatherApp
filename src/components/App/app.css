body, html {
    height: 100%;
}

.mainContainer {
    height: 100vh;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
}

.content {
    padding-top: 80px;
    height: 100vh;
    overflow-y: auto;
    padding-bottom: 40px;
}
