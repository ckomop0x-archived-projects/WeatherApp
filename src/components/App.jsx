import React from 'react';
import { HashRouter as Router, Route, Link, Switch } from 'react-router-dom';
import Header from './Header/index';
import Footer from './Footer/index';
import { PLACES } from '../helpers/places';
import Home from './Home';
import './app.css';
import CitySelector from './CitySelector';

const App = () => {
  return (
    <Router>
      <div className="mainContainer">
        <Header title="Weather Around the World"/>
        <div className="container content">
          <h3>Select a city</h3>
          <div>
            {PLACES.map((place, index) => (
                <Link key={index} to={`${place.name}`}>
                  <button type="button" className="btn btn-light" key={index}>
                    {place.name}
                  </button>
                </Link>
              )
            )}
          </div>
          {/*<CitySelector/>*/}
          <Switch>
            <Route exact path="/" render={() => <Home city='amsterdam'/>}/>
            <Route path="/:city" render={({match}) => <Home city={match.params.city}/>}/>
          </Switch>
        </div>
        <Footer footerTitle="&copy; 2017-2019 Weather app by Pavel Ckomop0x Klochkov."/>
      </div>
    </Router>
  );
};

export default App;
