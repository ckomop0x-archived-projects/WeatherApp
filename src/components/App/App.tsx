import React, { FunctionComponent } from 'react'
import { HashRouter as Router, Route, NavLink, Switch } from 'react-router-dom'
import { Header } from '../../components'
import Footer from '../Footer'
import { PLACES } from '../../helpers/places'
import Home from '../Home'
import { styled } from '@material-ui/styles'
import Button from '@material-ui/core/Button'
import { useTranslation } from 'react-i18next'
import CssBaseline from '@material-ui/core/CssBaseline'
import { version } from '../../../package.json'
import './app.css'

const ButtonStyled = styled(Button)({
  background: 'linear-gradient(45deg, #FE6B8B 30%, #FF8E53 90%)',
  border: 0,
  borderRadius: 3,
  boxShadow: '0 3px 5px 2px rgba(255, 105, 135, .3)',
  color: 'white',
  height: 24,
  padding: '0 10px',
})

const NavLinkStyled = styled(NavLink)({
  textDecoration: 'none',
})

const App: FunctionComponent = () => {
  const { t, i18n } = useTranslation()

  const changeLanguage = (lng: string) => {
    i18n.changeLanguage(lng)
  }

  return (
    <React.Fragment>
      <CssBaseline />
      <Router>
        <div className="mainContainer">
          <Header title={t('page.title')} changeLanguage={changeLanguage} />
          <div className="container content">
            <h3>{t('Select a city')}</h3>
            <div>
              {PLACES.map((place, index) => (
                <NavLinkStyled key={index} to={`${place.name}`}>
                  <ButtonStyled key={index}>
                    {/*<button type="button" className="btn btn-light" key={index}>*/}
                    {place.name}
                    {/*</button>*/}
                  </ButtonStyled>
                </NavLinkStyled>
              ))}
            </div>
            <Switch>
              <Route exact path="/" render={() => <Home city="amsterdam" />} />
              <Route path="/:city" render={({ match }) => <Home city={match.params.city} />} />
            </Switch>
          </div>
          <Footer footerTitle={`&copy; 2017-2019 Weather app by Pavel "Ckomop0x" Klochkov. v${version}`} />
        </div>
      </Router>
    </React.Fragment>
  )
}

export default App
