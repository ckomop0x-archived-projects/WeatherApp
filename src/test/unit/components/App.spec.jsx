import React from 'react';
import renderer from 'react-test-renderer';
import App from '../../../components/App';


describe('Components', () => {
    describe('App', () => {
        it('renders without crashing', () => {
            const wrapper = renderer.create(<App/>).toJSON();
            expect(wrapper).toMatchSnapshot();
        });
    });
});

