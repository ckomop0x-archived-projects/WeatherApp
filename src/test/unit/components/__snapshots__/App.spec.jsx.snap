// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components App renders without crashing 1`] = `
<div
  className="mainContainer"
>
  <header
    className="MuiPaper-root-11 MuiPaper-elevation4-17 MuiAppBar-root-2 MuiAppBar-positionFixed-3 MuiAppBar-colorPrimary-9 mui-fixed Header-header-1"
  >
    <div
      className="MuiToolbar-root-38 MuiToolbar-regular-40 MuiToolbar-gutters-39"
    >
      <h1
        className="MuiTypography-root-42 MuiTypography-headline-47 MuiTypography-colorInherit-71"
      >
        Weather Around the World
      </h1>
    </div>
  </header>
  <div
    className="container content"
  >
    <h3>
      Select a city
    </h3>
    <div>
      <a
        href="#/Amsterdam"
        onClick={[Function]}
      >
        <button
          className="btn btn-light"
          type="button"
        >
          Amsterdam
        </button>
      </a>
      <a
        href="#/Rotterdam"
        onClick={[Function]}
      >
        <button
          className="btn btn-light"
          type="button"
        >
          Rotterdam
        </button>
      </a>
      <a
        href="#/Moscow"
        onClick={[Function]}
      >
        <button
          className="btn btn-light"
          type="button"
        >
          Moscow
        </button>
      </a>
      <a
        href="#/Kiev"
        onClick={[Function]}
      >
        <button
          className="btn btn-light"
          type="button"
        >
          Kiev
        </button>
      </a>
      <a
        href="#/Uzhhorod"
        onClick={[Function]}
      >
        <button
          className="btn btn-light"
          type="button"
        >
          Uzhhorod
        </button>
      </a>
      <a
        href="#/Prague"
        onClick={[Function]}
      >
        <button
          className="btn btn-light"
          type="button"
        >
          Prague
        </button>
      </a>
      <a
        href="#/Santa Cruz"
        onClick={[Function]}
      >
        <button
          className="btn btn-light"
          type="button"
        >
          Santa Cruz
        </button>
      </a>
      <a
        href="#/Honolulu"
        onClick={[Function]}
      >
        <button
          className="btn btn-light"
          type="button"
        >
          Honolulu
        </button>
      </a>
    </div>
    <div>
      <div
        className="col-md-12"
      >
        <div>
          Loading
          <sup>
            *
          </sup>
        </div>
      </div>
    </div>
  </div>
  <footer
    className="footer"
  >
    <div
      className="container"
    >
      Â© 2017-2019 Weather app by Pavel Ckomop0x Klochkov.
    </div>
  </footer>
</div>
`;
